### DISCUSSION TITLE
Is build still needed if I am building an application, not a JS library?

### DISCUSSION THREAD

**Discussion by HuakunShen** [oven-sh/bun#4968] [Q&A]
I have a expressjs server and is thinking about switching to use bun runtime. 
Before, I had to compile TypeScript into JavaScript and run with nodejs.

I know bun is a bundler, but since bun can directly run TypeScript files, I don't know if it's still necessary to bundle into JavaScript.

I tested bun with a simple algorithm script, running TypeScript is much faster than nodejs running JS. 
Running JS or TS with Bun took the same amount of time. But this experiment didn't involve importing from different libraries.

I don't know if import from tons of libraries will slow down. Is bundling into JS and run with bun still a better choice in terms of speed and memory consumption?

I had some problem running the bundled JS file, that's why I am asking.

**Comment by gudh** [comment]
Same question here.

Is it slower for Bun to run typescript directly than the compiled javascript? Any benchmarks?

**Reply by SWATOPLUS** [reply]
It will be definitely slower to run ts instead of js. But need to understand how much slower.

But your still need code verification stage, to ensure that you don't have any type errors because of method name misprint.

**Reply by RiskyMH** [reply]
note that for bun:
* there really isn't any speed [difference](https://x.com/jarredsumner/status/1955540627179049471)*
* it also don't do any type checking

<sub> \* this is excluding bundling and minifying as they _can_ make the code faster to parse and then run</sub>
